### DISCUSSION TITLE
How to send 2 or more existing photos (by fileID) with text?

### DISCUSSION THREAD

**Discussion by resssoft** [go-telegram-bot-api/telegram-bot-api#432] [Q&A] [answered]
NewMessage - send text only
NewPhotoShare - send one photo by fileID (without text)
NewMediaGroup - 2 or more new files (without text)
...
?

**Comment by Syfaro** [comment, marked as answer]
Each `InputMediaPhoto` has a `Caption` field which can be set to arbitrary text. Telegram clients only show a caption from a media group below the media if exactly one element has a caption set, official clients seem to set the first photo's caption.
